it/src/test/scala/com/wavesplatform/it/sync/matcher/smartcontracts/OrdersFromScriptedAssetTestSuite.scala

```diff
@@ -0,0 +1,234 @@
+package com.wavesplatform.it.sync.matcher.smartcontracts
+
+import com.typesafe.config.{Config, ConfigFactory}
+import com.wavesplatform.account.{AddressScheme, PrivateKeyAccount}
+import com.wavesplatform.it.api.SyncMatcherHttpApi._
+import com.wavesplatform.it.matcher.MatcherSuiteBase
+import com.wavesplatform.it.sync.matcher.config.MatcherDefaultConfig
+import com.wavesplatform.it.sync.matcher.config.MatcherPriceAssetConfig.createSignedIssueRequest
+import com.wavesplatform.it.sync.matcherFee
+import com.wavesplatform.it.util._
+import com.wavesplatform.lang.v1.compiler.Terms
+import com.wavesplatform.state.EitherExt2
+import com.wavesplatform.transaction.assets.{IssueTransactionV1, IssueTransactionV2}
+import com.wavesplatform.transaction.assets.exchange.{AssetPair, Order, OrderType}
+import com.wavesplatform.transaction.smart.script.v1.ScriptV1
+import com.wavesplatform.it.api.SyncHttpApi.NodeExtSync
+import com.wavesplatform.transaction.smart.script.{Script, ScriptCompiler}
+
+import scala.util.Random
+
+/**
+  * Rules:
+  * 1. If the script fails during placing, the matcher must reject an order
+  * 2. If the script fails during execution, the matcher must cancel both orders (submitted and counter)
+  */
+class OrdersFromScriptedAssetTestSuite extends MatcherSuiteBase {
+
+  import OrdersFromScriptedAssetTestSuite._
+
+  override protected def nodeConfigs: Seq[Config] = Configs
+
+  "can place if the script returns TRUE" in {
+    val pair = AssetPair.createAssetPair(UnscriptedAssetId, AllowAssetId).get
+    val counterId =
+      matcherNode.placeOrder(matcherPk, pair, OrderType.SELL, 100000, 2 * Order.PriceConstant, version = 2, fee = smartTradeFee).message.id
+    matcherNode.waitOrderStatus(pair, counterId, "Accepted")
+    matcherNode.cancelOrder(matcherPk, pair, counterId)
+  }
+
+  "can't place if the script returns FALSE" in {
+    val pair = AssetPair.createAssetPair(UnscriptedAssetId, DenyAssetId).get
+    val order = Order.buy(
+      matcherPk,
+      matcherPk,
+      pair,
+      100000,
+      2 * Order.PriceConstant,
+      timestamp = System.currentTimeMillis(),
+      expiration = System.currentTimeMillis() + 60 * 60 * 30,
+      matcherFee = smartTradeFee,
+      version = 2
+    )
+    matcherNode.expectIncorrectOrderPlacement(order, 400, "OrderRejected")
+  }
+
+  "can execute against unscripted, if the script returns TRUE" in {
+    markup("place counter")
```

Comment:
Better to use `info()` or `Given()/When()/Then()/And()` (extend the class with `GivenWhenThen`).
The `markup()` is more intended for HTML reports.

Reply:
Copied this from your RestOrderLimitTestSuite :D